executor/executor.go

```diff
@@ -1360,7 +1360,18 @@ func (e *SelectionExec) Next() (*Row, error) {
 			return nil, errors.Trace(err)
 		}
 		if match {
-			return srcRow, nil
+			row := &Row{
+				RowKeys: srcRow.RowKeys,
+				Data:    make([]types.Datum, 0, len(srcRow.Data)),
+			}
+			for _, v := range e.Schema() {
```

Comment:
Before projection elimination, schema of Selection always equals to schema of DataSource,
but when we eliminate the Projection, schema of Selection may be changed, we can not 
return the data retrieved from Datasource directly to the parent operator.